r-package/grf/tests/testthat/test_causal_forest.R

```diff
@@ -125,12 +125,105 @@ test_that("local linear causal forests with large lambda are equivalent to causa
    p = 6
    X = matrix(rnorm(n*p), n, p)
    W = rbinom(n, 1, 0.5)
-   TAU = 2*X[,1] + X[,2]
-   Y = W * TAU + rnorm(n)
+   tau= 2*X[,1] + X[,2]
+   Y = W * tau + rnorm(n)
 
    forest = causal_forest(X, Y, W, num.trees = 400)
    preds.ll = predict(forest, X, linear.correction.variables = 1:ncol(X), ll.lambda = 1e5)$predictions
    preds.cf = predict(forest)$predictions
 
    expect_true(mean((preds.ll - preds.cf)^2) < 0.02)
 })
+
+
+test_that("predictions are invariant to scaling of the sample weights.", {
+   n = 100
+   p = 6
+   X = matrix(rnorm(n*p), n, p)
+   e = 1/(1+exp(-2*X[,1]+3*X[,2]))
+   W = rbinom(n, 1, e)
+   tau = 2*X[,1] + X[,2]
+   Y = W * tau + rnorm(n)
+   e.cc = 1/(1+exp(-2*X[,1]))
+   sample.weights = 1/e.cc
+
+   forest.1 = causal_forest(X, Y, W, sample.weights = sample.weights, seed=1)
+   forest.2 = causal_forest(X, Y, W, sample.weights = 40*sample.weights, seed=1)
+   expect_true(all(abs(forest.2$predictions - forest.1$predictions) < 1e-10))
+})
+
+test_that("IPCC weighting in the training of a causal forest with missing data improves its complete-data MSE.", {
+   old.seed = .Random.seed
```

Comment:
It's either this, large sample sizes, or tests that fail on a single platform with fairly small probability.

Reply:
Of course we need to set the seed. What I'm asking about is why we need to do `old.seed = .Random.seed`, and everything that follows. As opposed to just putting `set.seed(1)` at the top of the test.

Reply:
Oh. I wanted to avoid setting the seed for every other test.  Maybe testthat takes care of that automatically, but I couldn't find any documentation suggesting that.

Reply:
Actually, just remembered, we currently set the seed at the top of every test file. So there shouldn't be a need to set the seed further inside any specific test.

Reply:
Thanks, hadn't noticed.